addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/__tests__/filesystem-role-service.test.js

```diff
@@ -0,0 +1,366 @@
+  accessType = 'readwrite',
+  envPermission = { read: true, write: true },
+  folder = '/',
+  kmsScope = 'none',
+} = {}) => ({
+  id,
+  category,
+  accountId,
+  awsPartition,
+  bucketAccess,
+  bucket,
+  qualifier,
+  appRoleArn,
+  accessType,
+  envPermission,
+  folder,
+  kmsScope,
+});
+
+const createAppRole = ({
+  arn = 'arn:aws:iam::684277579687:role/swb-IhsKhN8GsLneiis11ujlb8-app-1607537845811',
+  accountId = '1122334455',
+  mainRegion = 'us-east-1',
+  awsPartition = 'aws',
+  bucket = 'bucket-1',
+  bucketRegion = 'us-east-1',
+  status = 'pending',
+  name = 'swb-IhsKhN8GsLneiis11ujlb8-app-1607537845811',
+  qualifier = 'swb-IhsKhN8GsLneiis11ujlb8',
+  boundaryPolicyArn = 'arn:aws:iam::684277579687:policy/swb-IhsKhN8GsLneiis11ujlb8-app-1607537845811',
+  studies = {
+    'study-1': {
+      accessType: 'readonly',
+      kmsScope: 'none',
+      folder: '/',
+    },
+  },
+} = {}) => ({
+  accountId,
+  arn,
+  name,
+  mainRegion,
+  awsPartition,
+  bucket,
+  bucketRegion,
+  status,
+  qualifier,
+  boundaryPolicyArn,
+  studies,
+});
+
+const createAdminContext = ({ uid = 'uid-admin' } = {}) => ({
+  principalIdentifier: { uid },
+  principal: { isAdmin: true, userRole: 'admin', status: 'active' },
+});
+
+describe('DataSourceBucketService', () => {
+  let container;
+  let service;
+  let appRoleService;
+  let usageService;
+
+  beforeEach(async () => {
+    // Initialize services container and register dependencies
+    container = new ServicesContainer();
+
+    container.register('dbService', new DbService());
+    container.register('jsonSchemaValidationService', new JsonSchemaValidationService());
+    container.register('settings', new SettingsService());
+    container.register('aws', new Aws());
+    container.register('log', new Logger());
+    container.register('pluginRegistryService', new PluginRegistryService());
+    container.register('authorizationService', new AuthService());
+    container.register('auditWriterService', new AuditService());
+    container.register('roles-only/applicationRoleService', new ApplicationRoleService());
+    container.register('resourceUsageService', new ResourceUsageService());
+    container.register('roles-only/filesystemRoleService', new FilesystemRoleService());
+    await container.initServices();
+
+    service = await container.find('roles-only/filesystemRoleService');
+    appRoleService = await container.find('roles-only/applicationRoleService');
+    usageService = await container.find('resourceUsageService');
+  });
+
+  describe('when allocating a role', () => {
+    it('skip if study bucket access is not roles', async () => {
+      const uid = 'u-currentUserId';
+      const requestContext = { principalIdentifier: { uid } };
+      const study = { bucketAccess: 'something else' };
+      const env = {};
+      const accountId = '1234456789012';
+
+      usageService.getResourceUsage = jest.fn(() => {
```

Comment:
Nit: This can be done by `expect(usageService.getResourceUsage).not.toHaveBeenCalled();`

Reply:
Oops, sorry, this was not meant to be here. I did that while I was doing some debugging and forget to remove it.